Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.SwitchExpression.cs

```diff
@@ -49,18 +50,22 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public SwitchExpression Update (Expression switchValue, IEnumerable<SwitchCase> cases, Expression defaultBody)
     {
+      Contract.Requires(switchValue != null);
```

Comment:
I'm worried adding too many conditions will result in unprovable contract warnings in any code trying to use the API. I'm not sure what to do about it. I wish we could use `Contract.Assume` in a precondition block to indicate conditions that are true but shouldn't attempt to be proven. This would allow us to show the conditions in the editor extensions (thus helping developers do things right) without negatively impacting the usability of cccheck. I'll let you decide which of the conditions we identified should be added, and which ones should be omitted.